keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,301 @@
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class AugMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Performs the AugMix data augmentation technique.
+
+    AugMix mixes several chains of augmentations together using weights sampled
+    from a Dirichlet distribution.  A single chain consist of a series of
+    individual augmentations.  The resultant image is further mixed with the
+    original image to form the final augmented image.
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written (low, high).
+            This is typically either `(0, 1)` or `(0, 255)` depending
+            on how your preprocessing pipeline is setup.
+        severity: A tuple of two floats, a single float or a `keras_cv.FactorSampler`.
+            A value is sampled from the provided range.  If a float is passed, the
+            range is interpreted as `(0, severity)`. This value represents the
+            level of strength of augmentations and is in the range [0, 1].
+            Defaults to 0.3.
+        width: an integer representing the number of different chains to
+            be mixed. Defaults to 3.
+        depth: an integer representing the number of transformations in
+            the chains. A negative value enables stochastic depth uniformly
+            in [1,3]. Defaults to -1.
+        alpha: a float value used as the probability coefficients for the
+            Beta and Dirichlet distributions. Defaults to 1.0.
+        seed: Integer. Used to create a random seed.
+
+    References:
+        [AugMix paper](https://arxiv.org/pdf/1912.02781)
+        [Official Code](https://github.com/google-research/augmix)
+        [Unoffial TF Code](https://github.com/szacho/augmix-tf)
+
+    Sample Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    augmix = keras_cv.layers.preprocessing.mix_up.AugMix([0, 255])
+    augmented_images = augmix(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        severity=0.3,
+        width=3,
+        depth=-1,
+        alpha=1.0,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, **kwargs)
+        self.value_range = value_range
+        self.width = width
+        self.depth = depth
+        self.alpha = alpha
+        self.seed = seed
+        self.auto_vectorize = False
+        self.severity = severity
+        self.severity_factor = preprocessing.parse_factor(
+            self.severity, min_value=0.01, param_name="severity", seed=self.seed
+        )
+
+        # initialize layers
+        self.auto_contrast = layers.AutoContrast(value_range=self.value_range)
+        self.equalize = layers.Equalization(value_range=self.value_range)
+
+    @staticmethod
+    def _sample_from_dirichlet(alpha):
+        gamma_sample = tf.random.gamma(shape=(), alpha=alpha)
```

Comment:
Should we create a gamma sampler in core and use it here?  This seems exactly in scope of the sampler

Reply:
We can do that, yes. Gamma distribution is in every interpolative augmentation technique I've seen thus far. This can be used to implement the beta distribution in MixUp, FourierMix, and AugMix as well. I think I can work on this after the merge of this PR and use it to sample for other augmentations as part of a follow-up PR.